src/main.js

```diff
@@ -154,3 +154,15 @@ Sidebar.prototype.close = function() {
 Object.assign(window.OCA.Files, {
 	Sidebar: new Sidebar(),
 })
+
+if (!window.OCA.Talk) {
+	window.OCA.Talk = {}
+}
+window.OCA.Talk.registerMessageAction = (label, callback, icon) => {
```

Comment:
JSdoc annotations for this and especially for the callback parameters would be nice

Reply:
Do we also need a "works on this message" call back? so we can have apps that allow interacting with e.g. system messages, old messages, etc. and others that don't (similar to deletion only working 6 hours)

Reply:
added comments!